sdk/src/azure/platform/az_posix.c

```diff
@@ -26,3 +57,165 @@ AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds)
   (void)usleep((useconds_t)milliseconds * _az_TIME_MICROSECONDS_PER_MILLISECOND);
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_platform_get_random(int32_t* out_random)
+{
+  _az_PRECONDITION_NOT_NULL(out_random);
+
+  *out_random = (int32_t)random();
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_platform_timer_create(
+    _az_platform_timer* timer_handle,
+    _az_platform_timer_callback callback,
+    void* sdk_data)
+{
+  _az_PRECONDITION_NOT_NULL(timer_handle);
+  _az_PRECONDITION_NOT_NULL(callback);
+
+  // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
+  memset(timer_handle, 0, sizeof(_az_platform_timer));
+
+  timer_handle->_internal.platform_timer._internal.callback = callback;
+  timer_handle->_internal.platform_timer._internal.sdk_data = sdk_data;
+
+  timer_handle->_internal.sev.sigev_notify = SIGEV_THREAD;
+  timer_handle->_internal.sev.sigev_notify_function = &_timer_callback_handler;
+  timer_handle->_internal.sev.sigev_value.sival_ptr = timer_handle;
+
+  if (0
+      != timer_create(
+          _az_PLATFORM_POSIX_CLOCK_ID,
+          &timer_handle->_internal.sev,
+          &timer_handle->_internal.timerid))
+  {
+    if (EAGAIN == errno)
```

Comment:
I don't think any of these are actionable. We could log the error for debugging but why would we convert from one type of error to another if our code doesn't use these return codes (which will need to be the same on Windows, Linux, Mac, Embedded RTOSes, etc)?

Please add _only_ error codes that will be generic across PALs and _must_ be used by our code.